src/Facets/GenericSwapFacet.sol

```diff
@@ -7,38 +10,261 @@ import { ReentrancyGuard } from "../Helpers/ReentrancyGuard.sol";
+        IERC20(_swapData.sendingAssetId).safeTransferFrom(
+            msg.sender,
+            address(this),
+            _swapData.fromAmount
+        );
+
+        // ensure that contract (callTo) and function selector are whitelisted
+        if (
+            // !((LibAsset.isNativeAsset(_swapData.sendingAssetId) || // TODO: confirm that we can safely remove this
+            // LibAllowList.contractIsAllowed(_swapData.approveTo)) && // TODO: confirm that we can safely remove this
+            !(LibAllowList.contractIsAllowed(_swapData.callTo) &&
+                LibAllowList.selectorIsAllowed(bytes4(_swapData.callData[:4])))
+        ) revert ContractCallNotAllowed();
+
+        // get initial token balance
+        uint256 initialBalance = address(this).balance;
+
+        // approve DEX to spend tokens
+        IERC20(_swapData.sendingAssetId).approve(
+            _swapData.approveTo,
+            _swapData.fromAmount
+        );
+
+        // execute swap
+        // solhint-disable-next-line avoid-low-level-calls
+        (bool success, bytes memory res) = _swapData.callTo.call(
+            _swapData.callData
+        );
+        if (!success) {
+            string memory reason = LibUtil.getRevertMsg(res);
+            revert(reason);
+        }
+
+        // get final balance
+        uint256 amountReceived = address(this).balance - initialBalance;
+
+        // ensure that minAmountOut was received
+        if (amountReceived < _minAmountOut)
+            revert CumulativeSlippageTooHigh(_minAmountOut, amountReceived);
+
+        // transfer funds to receiver
+        // solhint-disable-next-line avoid-low-level-calls
+        (success, ) = _receiver.call{ value: amountReceived }("");
+        if (!success) revert NativeAssetTransferFailed();
+
+        // emit event
+        emit LiFiGenericSwapCompleted(
+            _transactionId,
+            _integrator,
+            _referrer,
+            _receiver,
+            _swapData.sendingAssetId,
+            _swapData.receivingAssetId,
+            _swapData.fromAmount,
+            amountReceived
+        );
+    }
+
+    /// @notice Performs a single swap from the network's native token to ERC20 token
+    /// @param _transactionId the transaction id associated with the operation
+    /// @param _integrator the name of the integrator
+    /// @param _referrer the address of the referrer
+    /// @param _receiver the address to receive the swapped tokens into (also excess tokens)
+    /// @param _minAmountOut the minimum amount of the final asset to receive
+    /// @param _swapData an object containing swap related data to perform swaps before bridging
+    function swapTokensSingleNativeToERC20(
+        bytes32 _transactionId,
+        string calldata _integrator,
+        string calldata _referrer,
+        address payable _receiver,
+        uint256 _minAmountOut,
+        LibSwap.SwapData calldata _swapData
+    ) external payable refundExcessNative(_receiver) {
```

Comment:
I think it's still a good idea to keep this.

Reply:
ok, will do.
Do we need it for both cases that involve a native (ERC>Native , Native > ERC) or only for one of them?
@ezynda3